Collections_HW/homework2/collections/SortByFirstNameAndByAge.java

```diff
@@ -0,0 +1,50 @@
+package homework2.collections;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+// This class sorts by First Name in ascending order
+class SortByName implements Comparator<Student> {
```

Comment:
Consider having a separate file for each of three classes.

Reply:
SortByName is a comparator and by itself doesn't sort. Consider renaming into AgeDescendingComparator or give  another descriptive name.

Reply:
Thank you! I was thinking of separating file, but I wasn't sure.